### PR TITLE
Additional code to handle discarded changes

### LINKED ISSUE #9361: "Discard changes" in "Save before closing" should delete "newer" backups
As user, I work on JabRef. Then, I add some things. I maybe want to cancel my changes. Example:

grafik

Then, at the start of JabRef, JabRef should NOT prompt me to restore these changes.

Four options:

- Option 1: Delete newer backup files
- Option 2: Rename backup files to files with certain keywords (and delete them after some time)
- Option 3: Make backups appearing "older" (we need to fake time stamps)
- Option 4: Create `discarded` file to mark that the user discarded changes. On start of JabRef, JabRef removes that file and backup takes normally place. Example filename: `269b5589--test.bib--discarded.bak`. When JabRef starts freshly, it creates new backups in the backup manner. After a normal save and an additional restart, JabRef MUST NOT assume that the last quit was a discard.

We opt for Option 4.

### PR DESCRIPTION
**Pull request by mmohanb1**
Fix #9361

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by HoussemNasri** [comment]
Hello! Welcome to JabRef!

I have read through the changes made in this PR and I'm not entirely sure what is the underlying issue you are addressing here. If you could provide more information or context, that would be great. If this PR is in response to an issue that's already been reported, please include a link to the issue in the PR description. If this is a new issue, please describe it here and mention your motivations for addressing it.

File: File not found [hunk 1]
```diff
@@ -0,0 +1,7 @@
+@Article{,
+  author = {Test1-abc},
+  title  = {Test1},
+  month  = may,
+}
+
+@Comment{jabref-meta: databaseType:bibtex;}
```
Comment (koppor) [inline File not found:7]: The path of the file is very strange. I think you accidently commited the file and it should not be part of the PR?

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 2]
```diff
@@ -1200,6 +1202,29 @@ private boolean confirmClose(LibraryTab libraryTab) {
...
             return false;
         }
+        if (response.isPresent() && response.get().equals(discardChanges)) {
+            //filename = filename + "--discarded.bak";
+
+            Path directory = Path.of(filename);
+            try {
+                SaveDatabaseAction saveAction = new SaveDatabaseAction(libraryTab, prefs, Globals.entryTypesManager);
+                String backUpPath = BackupFileUtil.getPathOfLatestExisingBackupFile(directory, BackupFileType.BACKUP).map(Path::toString).
+                        orElse(Localization.lang("File not found"));
```
Comment (koppor) [inline src/main/java/org/jabref/gui/JabRefFrame.java:1212]: There should not be a file `File not found` be created.

File: src/main/java/org/jabref/logic/autosaveandbackup/BackupManager.java [hunk 3]
```diff
@@ -189,7 +198,8 @@ private Optional<Path> determineBackupPathForNewBackup() {
      * @param backupPath the path where the library should be backed up to
      */
     private void performBackup(Path backupPath) {
-        if (!needsBackup) {
+
+        if (!needsBackup || discardedFileExists) {
```
Comment (koppor) [inline src/main/java/org/jabref/logic/autosaveandbackup/BackupManager.java:202]: Why the check here? A backup should be performed regardless of the discared file. The discarded file was of the **last** JabRef run.

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 4]
```diff
@@ -1200,6 +1202,29 @@ private boolean confirmClose(LibraryTab libraryTab) {
             // Save was cancelled or an error occurred.
             return false;
         }
+        if (response.isPresent() && response.get().equals(discardChanges)) {
+            //filename = filename + "--discarded.bak";
+
+            Path directory = Path.of(filename);
+            try {
+                SaveDatabaseAction saveAction = new SaveDatabaseAction(libraryTab, prefs, Globals.entryTypesManager);
```
Comment (koppor) [inline src/main/java/org/jabref/gui/JabRefFrame.java:1210]: Do NOT Call UI methods for saving. The logic for creating a `--discared` file should be in the BackupManager. You can see the saving at org.jabref.logic.autosaveandbackup.BackupManager#performBackup.

As written in https://github.com/JabRef/jabref/issues/9361#issuecomment-1343957665, it really should be done in BackupManager.

File: src/main/java/org/jabref/gui/JabRefFrame.java [hunk 5]
```diff
@@ -1200,6 +1202,29 @@ private boolean confirmClose(LibraryTab libraryTab) {
...
+                String backUpPath = BackupFileUtil.getPathOfLatestExisingBackupFile(directory, BackupFileType.BACKUP).map(Path::toString).
+                        orElse(Localization.lang("File not found"));
+                backUpPath = backUpPath.replace(".bak","--discarded.bak");
+
+                if (saveAction.saveAs(Path.of(backUpPath))) {
+                    BackupManager.setDiscardedFileExists(true);
+                    return true;
+                }
```
Comment (koppor) [inline src/main/java/org/jabref/gui/JabRefFrame.java:1218]: I think, you misunderstood the user interaction of JabRef.

1. User starts JabRef
2. User opens a file `demo.bib`
3. User changes  file `demo.bib` for 10 minutes (and does not save)
5. User closes file.
6. JabRef diplays "File changed dialog"
7. User selects "Discard changes"
8. User opens the file `demo.bib` again
9. JabRef should NOT prompt for restoring backup

In the code at hand, sets the flag after step 7, but not at step 8, where it is needed.

**Comment by koppor** [comment]
Superseeded by https://github.com/JabRef/jabref/pull/9457
